server/routes/caseNotes/view/caseNotePresenter.ts

```diff
@@ -13,10 +13,15 @@ export default class CaseNotePresenter {
   constructor(
     private caseNote: CaseNote,
     private sentByUserName: string,
-    public loggedInUserType: 'service-provider' | 'probation-practitioner'
+    public loggedInUserType: 'service-provider' | 'probation-practitioner',
+    private backlinkPageNumber: number | null
   ) {}
 
-  backLinkHref = `/${this.loggedInUserType}/referrals/${this.caseNote.referralId}/case-notes`
+  // There is the chance that further case notes are added which pushes the selected case note onto another page.
+  // The backlink will now take them to the page they came from, but potentially not the page the case note now lives.
+  backLinkHref = `/${this.loggedInUserType}/referrals/${this.caseNote.referralId}/case-notes${
```

Comment:
sorry to be picky on this - i think this should explicitly be checking for null instead of trying to cast the type `number | null` to number. it's either null or already a number. if you cast null to number you get NaN.